Jackson dependency subject to CVE-2016-3720

## Description
docker-client 8.4.0 uses a version of Jackson that is subject to a CVE. [Information can be found here.](https://nvd.nist.gov/vuln/detail/CVE-2016-3720)

[Additionally, there is a Medium severity issue with Rancher Labs.](https://nvd.nist.gov/vuln/detail/CVE-2017-7297)

## How to reproduce
Build with a system that exposes CVEs. My organization builds with Jenkins, using a maven plugin that lists CVEs.

## What do you expect
Use an updated version of Jackson that is not subject to the CVE.

## What happened instead
CVE.

## Software:
- Spotify's docker-client version: 8-4-0

## Full backtrace

C:/tools/apache-maven-3.3.9/localRepository/com/fasterxml/jackson/module/jackson-module-jaxb-annotations/2.6.0/jackson-module-jaxb-annotations-2.6.0.jar , CVE-2016-3720 , Severity: High

XML external entity (XXE) vulnerability in XmlMapper in the Data format extension for Jackson (aka jackson-dataformat-xml) allows attackers to have unspecified impact via unknown vectors.

C:/tools/apache-maven-3.3.9/localRepository/com/spotify/docker-client/8.4.0/docker-client-8.4.0.jar , CVE-2017-7297 , Severity: Medium
Rancher Labs rancher server 1.2.0+ is vulnerable to authenticated users disabling access control via an API call. This is fixed in versions rancher/server:v1.2.4, rancher/server:v1.3.5, rancher/server:v1.4.3, and rancher/server:v1.5.3.

```
$ git show-ref HEAD
f00168ec5c92e20272302415d70fb5633f655894 refs/remotes/origin/HEAD

$ mvn org.owasp:dependency-check-maven:check
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building docker-client 8.6.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- dependency-check-maven:1.4.5:check (default-cli) @ docker-client ---
[INFO] Checking for updates
[INFO] Skipping NVD check since last check was within 4 hours.
[INFO] Check for updates complete (2012 ms)
[INFO] Analysis Started
[INFO] Finished Archive Analyzer (0 seconds)
[INFO] Finished File Name Analyzer (0 seconds)
[INFO] Finished Jar Analyzer (1 seconds)
[INFO] Finished Central Analyzer (1 seconds)
[ERROR] ----------------------------------------------------
[ERROR] .NET Assembly Analyzer could not be initialized and at least one 'exe' or 'dll' was scanned. The 'mono' executable could not be found on the path; either disable the Assembly Analyzer or configure the path mono. On some systems mono-runtime and mono-devel need to be installed.
[ERROR] ----------------------------------------------------
[INFO] Finished Dependency Merging Analyzer (0 seconds)
[INFO] Finished Version Filter Analyzer (0 seconds)
[INFO] Finished Hint Analyzer (0 seconds)
[INFO] Created CPE Index (2 seconds)
[INFO] Finished CPE Analyzer (3 seconds)
[INFO] Finished False Positive Analyzer (0 seconds)
[INFO] Finished Cpe Suppression Analyzer (0 seconds)
[INFO] Finished NVD CVE Analyzer (0 seconds)
[INFO] Finished Vulnerability Suppression Analyzer (0 seconds)
[INFO] Finished Dependency Bundling Analyzer (0 seconds)
[INFO] Analysis Complete (9 seconds)
[WARNING]

One or more dependencies were identified with known vulnerabilities in docker-client:

jackson-core-2.6.0.jar (com.fasterxml.jackson.core:jackson-core:2.6.0, cpe:/a:fasterxml:jackson:2.6.0) : CVE-2016-7051
jackson-jaxrs-base-2.6.0.jar (com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:2.6.0, cpe:/a:fasterxml:jackson:2.6.0) : CVE-2016-7051, CVE-2016-3720


See the dependency-check report for more details.


[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 16.209 s
[INFO] Finished at: 2017-05-25T15:18:08-04:00
[INFO] Final Memory: 27M/780M
[INFO] ------------------------------------------------------------------------
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -40,7 +40,7 @@ The following methods are deprecated and will be removed in a future release:
 
 ### Upgraded dependencies
 
-Jackson has been upgraded from 2.6.0 to 2.8.6.
+Jackson has been upgraded from 2.6.0 to 2.8.8.
 
 ## 8.5.0 (released May 18 2017)
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -38,6 +38,10 @@ The following methods are deprecated and will be removed in a future release:
 
 [740](https://github.com/spotify/docker-client/issues/740), [759](https://github.com/spotify/docker-client/pull/759)
 
+### Upgraded dependencies
+
+Jackson has been upgraded from 2.6.0 to 2.8.6.
+
 ## 8.5.0 (released May 18 2017)
 
 ### Removal of deprecated methods

--- a/pom.xml
+++ b/pom.xml
@@ -89,17 +89,17 @@
     <dependency>
       <groupId>com.fasterxml.jackson.jaxrs</groupId>
       <artifactId>jackson-jaxrs-json-provider</artifactId>
-      <version>2.6.0</version>
+      <version>2.8.6</version>
     </dependency>
     <dependency>
       <groupId>com.fasterxml.jackson.datatype</groupId>
       <artifactId>jackson-datatype-guava</artifactId>
-      <version>2.6.0</version>
+      <version>2.8.6</version>
     </dependency>
     <dependency>
       <groupId>com.fasterxml.jackson.core</groupId>
       <artifactId>jackson-databind</artifactId>
-      <version>2.6.0</version>
+      <version>2.8.6</version>
     </dependency>
     <dependency>
       <groupId>org.glassfish.jersey.core</groupId>

--- a/pom.xml
+++ b/pom.xml
@@ -89,17 +89,17 @@
     <dependency>
       <groupId>com.fasterxml.jackson.jaxrs</groupId>
       <artifactId>jackson-jaxrs-json-provider</artifactId>
-      <version>2.8.6</version>
+      <version>2.8.8</version>
     </dependency>
     <dependency>
       <groupId>com.fasterxml.jackson.datatype</groupId>
       <artifactId>jackson-datatype-guava</artifactId>
-      <version>2.8.6</version>
+      <version>2.8.8</version>
     </dependency>
     <dependency>
       <groupId>com.fasterxml.jackson.core</groupId>
       <artifactId>jackson-databind</artifactId>
-      <version>2.8.6</version>
+      <version>2.8.8</version>
     </dependency>
     <dependency>
       <groupId>org.glassfish.jersey.core</groupId>

